tests/test_logging.py

```diff
@@ -178,28 +178,36 @@ def setup_logging_config(cli_options: Options, file_option_value: Optional[str]
         # Reset stream buffer
         stream.truncate()
 
+    # Log file to pass in the options.
+    # This config option will be overridden by the other config in this test case.
+    old_log_file = os.path.join(tmpdir, "old_log_file.txt")
     path_logging_config_file1 = os.path.join(tmpdir, "logging_config1.yml")
     path_logging_config_file2 = os.path.join(tmpdir, "logging_config2.yml")
     path_logging_config_file3 = os.path.join(tmpdir, "logging_config3.yml")
 
     # Set --logging-config option on CLI only
     write_logging_config_file(path=path_logging_config_file1, formatter="AAA %(message)s")
     # Also assert that the other config options are ignored when logging_config is set.
-    setup_logging_config(cli_options=Options(logging_config=path_logging_config_file1, verbose=1), file_option_value=None)
+    setup_logging_config(
+        cli_options=Options(logging_config=path_logging_config_file1, verbose=1, log_file=old_log_file, log_file_level="4"),
```

Comment:
Could we also test that the warning was properly emitted ?

Reply:
@arnaudsjs the discussion is related to this comment.
I want to keep all the assertions in the code and also assert that warnings are properly emitted